src/Stack/Upload.hs

```diff
@@ -67,14 +84,26 @@ instance FromJSON (FilePath -> HackageCreds) where
 withEnvVariable :: Text -> IO Text -> IO Text
 withEnvVariable varName fromPrompt = lookupEnv (T.unpack varName) >>= maybe fromPrompt (pure . T.pack)
 
+maybeGetHackageKey :: RIO m (Maybe HackageKey)
+maybeGetHackageKey = fmap (HackageKey . T.pack) <$> (liftIO $ lookupEnv (T.unpack "HACKAGE_KEY"))
+
+loadAuth :: HasLogFunc m => Config -> RIO m HackageAuth
+loadAuth config = do
+  maybeHackageKey <- maybeGetHackageKey
+  case maybeHackageKey of
+    Just key -> do
+      logInfo "HACKAGE_KEY found in env, using that for credentials."
```

Comment:
Personally, I don't think we need to log this. Seems like something added to aid development of this feature more than anything.

Reply:
I think it should stay, because if you forget to unset HACKAGE_KEY, you might get unexpected results.

Reply:
I agree that at first (and second) sight it looks very much like debug-level info, but I also think it should stay - for the reason Julian gave.

Would be worth reconsidering if this was the only line of output shown for this operation, but for `stack upload` there are at least 5 more log lines shown before it. These include `stack> '/tmp/stack-sdist-3ea4415731d1f1b8/source-files-list'`, which to me seems considerably farther from user interest than the `HACKAGE_KEY` var.

Maybe it would be a good candidate for a separate improvement in `stack upload` to clean up those lines, eg. grouping them into a separate log level.